airflow-core/src/airflow/models/variable.py

```diff
@@ -180,13 +180,11 @@ def get(
                 return var_val
 
     @staticmethod
-    @provide_session
     def set(
         key: str,
         value: Any,
         description: str | None = None,
         serialize_json: bool = False,
-        session: Session = None,
```

Comment:
Given `Variable.update` has a session, if we are doing to remove `@provide_session` I think this should be

`session: Session | None = None` -- that way we still use the provided session if there is one.

Reply:
Umm yeah makes sense. Let me do that

Reply:
Handled this in latest commit